src/algorithms/algorithm.h

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include <string>
+
+class Game;
+
+class Algorithm{
+private:
+    std::string name;
```

Comment:
I cannot think of any reason why wouldn't the class just override get_name() method

Reply:
Do you mean to delete this "name" field and just leave "get_name()" in every child class?

Reply:
Yeah, and return the name of the algorithm directly in the method. I would call it get_description() instead of get_name()

Reply:
Why would you call it "get_description()" if the only thing that it will return is its name?

Reply:
Ah, I've thought of adding parameters to the returned string

If u want to return name only get_name() is def better